test/merge-deps.js

```diff
@@ -0,0 +1,34 @@
+// @flow strict
+import {
+  mergeDeps,
+  type DeserializerGenerator,
+  type MetaType,
+  typeHeader,
+} from '../src/generator.js'
```

Comment:
I've been noodling on the "forgot to add an export to index.js" problem, and I think having the tests import from `index.js` may be the simplest thing to do for now.

Also, as a related issue, new tests need to be added to `test-suite.js`, which I can now see will have a similar problem of being forgotten. Mayhaps `test-suite.js` can be smartened up to slurp in all the files in the test directory, or a real test framework needs to be brought in.

Reply:
Re: tests - I'm thinking there's enough itch that a real framework would scratch it.

Re: `index.js` I think mandating that tests use that is very sensible.

Reply:
Something else I just noticed, `typeHeader` isn't currently exported by `../src/generator.js`.

Reply:
Good catch! Also I was hoping to do a big pass on `typeHeader` and making sure that type names with their type parameters are respected throughout the generation functions. But it'll probably be another pull request.

Reply:
I've exported `typeHeader` but still want to take a comprehensive pass at documenting it and using it everywhere in a later pull request.